Assets/Scripts/Hoop.cs

```diff
@@ -5,12 +5,60 @@
 public class Hoop : ABaseEntity
 {
     public int value = 1;
+    private int currentSpeed;
+    private Vector3 startPos;
+    private float r;
+    private float startHeight;
+    public int[] speeds = { 0, 5, 15, 30, 50 };
+    public int currentPattern = 0;
+    public Player player;
     public ScoreManager scoreManager;
+    public float vertMoveScale = 1.0f;
     void OnTriggerEnter(Collider other)
     {
         if (other.CompareTag("Basketball"))
         {
             scoreManager.Score += value;
         }
     }
+
+    public override void BaseStart()
+    {
+        Vector3 pos = new Vector3(player.transform.position.x, 0, player.transform.position.z);
+
+        startPos.Set(this.transform.position.x, 0, this.transform.position.z);
+        currentSpeed = speeds[0];
+        r = Vector3.Distance(startPos, pos);
+        startHeight = this.transform.position.y;
+    }
+    public override void BaseUpdate()
+    {
+        if (scoreManager.Difficulty <= speeds.Length && currentSpeed != speeds[scoreManager.Difficulty])
+        {
+            currentSpeed = speeds[scoreManager.Difficulty];
+            currentPattern++;
+            if (currentPattern == 3)

```

Comment:
What is current pattern 3?

Reply:
So this is the byproduct of trying to make code more general but implementing it with more specialized code. Current pattern is meant to store what kind of movement pattern the hoop has. The way I've implemented it has made it basically the same as the difficulty, unless you change it in the editor to be higher. I've left room for it to grow if we decide to make more patterns for the hoop to move. Right now, 1 will make the hoop move in a circle and 3 will make it move in a sine wave around the player. It'll default to the next lowest pattern if there's nothing defined for the current pattern (i.e. 2 will act like 1, anything greater than or equal to 4 will act like 3).